jwcrypto/jwk.py

```diff
@@ -663,22 +670,37 @@ def _ec_pri(self, k, curve):
         return ec.EllipticCurvePrivateNumbers(self._decode_int(k['d']),
                                               self._ec_pub(k, curve))
 
-    def _okp_pub(self, k, curve):
-        if curve == 'Ed25519':
-            return ed25519.Ed25519PublicKey.from_public_bytes(k['x'])
-        if curve == 'Ed2448':
-            return ed448.Ed448PublicKey.from_public_bytes(k['x'])
-        else:
-            raise NotImplementedError
-
-
-    def _okp_pri(self, k, curve='Ed25519'):
-        if curve != 'Ed25519':
-            return ed25519.Ed25519PrivateKey.from_private_bytes(k['d'])
-        if curve != 'Ed448':
-            return ed448.Ed448PrivateKey.from_private_bytes(k['d'])
-        else:
-            raise NotImplementedError
+    def _okp_pub(self, k):
+        if k['crv'] == 'Ed25519':
+            return ed25519.Ed25519PublicKey.from_public_bytes(
+                base64url_decode(k['x']))
+        elif k['crv'] == 'Ed448':
+            return ed448.Ed448PublicKey.from_public_bytes(
+                base64url_decode(k['x']))
+        elif k['crv'] == 'X25519':
+            return x25519.X25519PublicKey.from_public_bytes(
+                base64url_decode(k['x']))
+        elif k['crv'] == 'X448':
+            return x448.X448PublicKey.from_public_bytes(
+                base64url_decode(k['x']))
+        # No support for other curves
+        raise NotImplementedError
+
+    def _okp_pri(self, k):
+        if k['crv'] == 'Ed25519':
+            return ed25519.Ed25519PrivateKey.from_private_bytes(
+                base64url_decode(k['x']))
```

Comment:
these (here and below) should probably be k['d'] ?

Reply:
Yes they should, thank you. I'll fix it.